lstchain/high_level/interpolate.py

```diff
@@ -0,0 +1,536 @@
+import numpy as np
+import logging
+import astropy.units as u
+from astropy.table import Table, QTable
+from astropy.io import fits
+from astropy.time import Time
+
+from lstchain.__init__ import __version__
+
+from pyirf.io.gadf import (
+    create_aeff2d_hdu,
+    create_energy_dispersion_hdu,
+    create_psf_table_hdu,
+)
+from pyirf.interpolation import (
+    interpolate_effective_area_per_energy_and_fov,
+    interpolate_energy_dispersion,
+    interpolate_psf_table,  # interpolate_rad_max
+)
+from scipy.spatial import Delaunay, distance
+from scipy.interpolate import griddata
+
+log = logging.getLogger(__name__)
+
+
+def interp_params(params_list, data):
+    """
+    From a given list of angular parameters, to be used for interpolation,
+    take values from a given data table/dict.
+
+    Returns the neccessary values with applied functions as need be for the
+    interpolation, for each parameter as a list.
+    """
+    mc_pars = []
+    if "ZEN_PNT" in params_list:
+        if isinstance(data["ZEN_PNT"], u.Quantity):
+            unit = data["ZEN_PNT"].unit
+        else:
+            unit = data.comments["ZEN_PNT"]
```

Comment:
From the FITS standard:

> . Nonetheless, software should not depend on units being
expressed in this fashion within a keyword comment, and soft-
ware should not depend on any string within square brackets in
a comment field containing a proper units string.

So we should not parse units from the comments. We should write a well-defined unit and attach that to the value when reading.

Reply:
The issue you linked has nothing to do with units in comments.

> or use string representation of the quantities?

No. We should store quantities like this:

```
unit = u.s
hdu.header['FOO'] = foo.to_value(unit), f'[{unit:fits}]'
```

And parse like this:
```
foo = u.Quantity(hdu.header['FOO'], unit=u.s
```

So writing the unit as comment for humans looking at the header but using the fixed unit for writing and reading.